libtiff/libtiff_ctypes.py

```
@@ -786,59 +831,54 @@ def read_tiles(self, dtype=np.uint8):
         num_irows = self.GetField("ImageLength")
         if num_irows is None:
             num_irows = 1
-        num_idepth = self.GetField("ImageDepth")
-        if num_idepth is None:
-            num_idepth = 1
-
-        if num_idepth == 1 and num_irows == 1:
-            # 1D
-            full_image = np.zeros((num_icols,), dtype=dtype)
-        elif num_idepth == 1:
-            # 2D
-            full_image = np.zeros((num_irows, num_icols), dtype=dtype)
-        else:
-            # 3D
-            full_image = np.zeros((num_idepth, num_irows, num_icols),
-                                  dtype=dtype)
-
-        tmp_tile = np.zeros((num_trows, num_tcols), dtype=dtype)
-        tmp_tile = np.ascontiguousarray(tmp_tile)
-        for z in range(0, num_idepth):
-            for y in range(0, num_irows, num_trows):
-                for x in range(0, num_icols, num_tcols):
-                    r = self.ReadTile(tmp_tile.ctypes.data, x, y, z, 0)
+        # this number includes extra samples
```

As for TileWidth, if TileLength tag is not defined, it should fail, instead of defaulting to 1.